### PR TITLE
Hemophagus, The Hemophage Revamp, Part Three: A Taste For Blood

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request

This is the third part of the Revamp, this time focusing on the relationship between Hemophages and what they consume.

What's changed?

<details>
<summary>Introducing Hemophage foods</summary>
  It's food made primarily out of blood, to allow for Hemophages to eat something solid, for once! There's a few recipes, all accessible from the crafting menu (in the new Hemophage Food tab) or through examining the item (when it needs to be microwaved), except for raw blood rice, which is just from mixing 10 units of rice with 20 units of blood.

The full list of food can be found in `hemophage_foods.dm`, I'm a bit lazy right now after working on this all day.

Here's what they look like! All sprites except the first and the last three are from Candlejax, I'm still looking for sprites for the noodles because I wasn't entirely satisfied with the ones I had at hand.

image

</details>

<details>
<summary>Added the Bloodshot drink, replacing the Bloody Mary's blood-regenerating supremacy</summary>
  
image

I love the way this one looks. The original sprite is from Crumpaloo, which I then heavily modified to suit what I wanted to go for. 

The lore behind this one is that it's a more accessible version of a glitterworld medicine aiming to help give their tumor all the nutrients it needs to stay alive. It's a lot less effective, but also cheaper, still effective, and also tasting relatively good, *especially* for Hemophages.

How to make it? Three parts blood, one part Bloody Mary and one part sugar, for five parts of Bloodshot, 

Then, for the changes: Bloody Mary is now 3 times slower at regenerating blood, while its original blood regeneration is available on the Bloodshot instead! The drink will also taste _awful_ for everyone that's not an Hemophage and that is at their full amount of blood (it'll taste decent for those that are in need of blood, however!)
</details>

<details>
<summary>Hemophages will no longer be able to freely eat or drink anything they want</summary>
  Everything that's not bloody food, or drinks that will help regenerate their blood, will no longer be tolerated by their tumor, which will instead force them to vomit instantly upon consumption. The intended way to be able to circumvent this is through the usage of the new Taste Suppressor reagent, which will allow them to eat anything that's not toxic to them, even if they will find absolutely no enjoyment from doing such whatsoever.

Yes, that also affects consuming medicine and the like, they will now have to use pills, patches or syringes instead. Basically, everything that isn't food, or directly drinking the reagent.

The drinking part comes from their liver, whereas the eating part comes from their stomach. Yes, that does mean that getting those replaced will also allow you to consume things normally, that _might_ be changed in the future if I feel like it's necessary, when I will properly implement organ corruption.
</details>

<details>
<summary>A new Taste Suppressor reagent was added to allow Hemophages to eat without vomiting</summary>
  It comes at the disadvantage of being unable to enjoy anything they eat. That's pretty much all there is to it.

The recipe is one part sodium, one part sulfur and one part water, at just about any temperature, for three parts of Taste Suppressor, but there's also a ghetto recipe for it, which is two parts table salt, one part sulfur and one part water, requiring it to be heated up to 300 K in order for the reaction to create three parts of Taste Suppressor and one part of Chlorine (left over from the salt).
</details>

#### What's left?

- [ ] Get new sprites for the noodles.
- [x] Get this test-merged to make sure that there's no unintended side-effects from this.

## How This Contributes To The Skyrat Roleplay Experience

Adding more depth to Hemophages is something I've been working on doing for a while, and this in particular is something I've been meaning to do for like over two months at this point.

It'll be beneficial in really anchoring the Hemophages in the setting, by making the players need to think a bit more about their casual roleplay actions regarding food, whilst also providing new and interesting ways for Hemophages to feed!

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image

</details>

## Changelog

:cl: GoldenAlpharex
add: Added eight new dishes to Hemophages, most of which can be either crafted via the Hemophage Food subcategory in the crafting menu, or through microwaving!
add: Added the Bloodshot drink, being the new racial drink for Hemophages! It restores blood at the rate Bloody Mary used to restore blood before this update. It's also the intended way to deal with bloodloss via drinks, which means it'll also work on those that aren't full on blood! It might gross you out if you don't need it, however.
add: Added the Taste Suppressor reagent, which allows you to eat food you might not like by making you completely unable to taste anything. It's not a great solution, but hey, it works, even for Hemophages!
balance: Bloody Mary now restores blood at a third of its previous speed.
balance: Hemophages can now only eat food and drink reagents (via ingestion) that will help their body recovering some blood volume. They now need to use the Taste Suppressor reagent to be able to eat everything else, at the cost of not really tasting anything they eat while it's in their system.
imageadd: Added some new sprites for the new Hemophage food and drinks!
/:cl:

### CODE REVIEW

**Comment by rosextango** [comment]
Only concern is does the bloody mary change affect non-hemophages?

**Comment by GoldenAlpharex** [comment]
> Only concern is does the bloody mary change affect non-hemophages?

Yes, and on purpose. You're now meant to make Bloodshot instead of Bloody Mary, for bloodloss. It's a much more balanced drink for it than Bloody Mary is, while it still retains a weaker blood regeneration regardless.

File: modular_skyrat/modules/customization/modules/food_and_drinks/food/hemophage_food.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	)
+	tastes = list("blood" = 5, "congealed blood" = 3, "pasta" = 1)
+	foodtypes = GRAIN | GORE | BLOODY
+
+
+/obj/item/food/hemophage/blood_curd
+	name = "blood curd"
+	desc = "Also known as 'blood tofu' or 'blood pudding,' this Yangyu delicacy looks to be made of congealed and cooked blood. It's soft and smooth, slightly chewy, and rich in riboflavin."
+	icon_state = "blood_curd"
+	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 5, /datum/reagent/blood = 15) // You don't get the most out of this from just eating it raw, just to promote making dishes out of it.
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/food_and_drinks/food/hemophage_food.dm]: mind formatting this like this other lists please

File: modular_skyrat/modules/customization/modules/food_and_drinks/food/hemophage_food.dm [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	desc = "If you see this, then something's gone very wrong and you should report it whenever you get the chance."
+	icon = 'modular_skyrat/master_files/icons/obj/food/hemophage_food.dmi'
+	foodtypes = GORE | BLOODY
+
+
+/obj/item/food/hemophage/blood_rice_pearl
+	name = "kessen shinju"
+	desc = "A fun finger food. Little clumps of sticky rice with a bit of ground pork and green onion, all soaked and rolled in fresh blood; giving it a crimson hue. Recommended to serve hot!"
+	icon_state = "blood_rice_pearl"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/blood = 10)
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/food_and_drinks/food/hemophage_food.dm]: same goes for the rest of these

File: modular_skyrat/modules/customization/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -491,6 +491,44 @@
...
+	color = "#a30000"
+	boozepwr = 20 // The only booze in it is Bloody Mary
+	taste_description = "blood filled to the brim with nutrients of all kinds"
+	glass_icon = 'modular_skyrat/master_files/icons/obj/drinks.dmi'
+	glass_icon_state = "bloodshot"
+	glass_name = "glass of bloodshot"
+	glass_desc = "The history of the 'Bloodshot' is based in a mix of flavor-neutral chems devised to help deliver nutrients to a Hemophage's tumorous organs. Due to the expense of the real thing and the clinical nature of it, this liquor has been designed as a 'improvised' alternative; though, still tasting like a hangover cure. It smells like iron, giving a clue to the key ingredient."
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED | REAGENT_BLOOD_REGENERATING
+
+#define BLOODSHOT_DISGUST 25
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/reagents/chemistry/reagents/alcohol_reagents.dm:614]: i get why this is done this way, but i still don't like it

**Comment by Mithrun** [comment]
My only criticism of this? Now that there's food for me as a hemophage also it sorta just feels feels like i'm back to being a human. Otherwise good! Sad that i can't drink alcohol anymore though..

File: code/__DEFINES/~skyrat_defines/reagents.dm [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/// This reagent is useful for blood regeneration. Useful for Hemophages.
+#define REAGENT_BLOOD_REGENERATING (1<<9)
```
Comment (Zonespace27) [inline code/__DEFINES/~skyrat_defines/reagents.dm]: this is totally gonna conflict, do you think you could move this to a `chemical_flags_skyrat` var, like was done for the forging trait?

File: code/__DEFINES/food.dm [hunk 5]
```diff
@@ -18,6 +18,7 @@
 #define ORANGES (1<<17)
 #define BUGS (1<<18)
 #define GORE (1<<19)
+#define BLOODY (1<<20) // SKYRAT EDIT - Hemophage Food
```
Comment (Zonespace27) [inline code/__DEFINES/food.dm:21]: i feel similar here but this one might be too engrained

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/__DEFINES/food.dm (hunk 5 above)
Comment (GoldenAlpharex) [reply code/__DEFINES/food.dm:21]: Yeah it sadly is a bit too engrained. I'll deal with the conflict when it happens, and if it's no longer possible to do it this way I'll find another way.

**Review by Zonespace27 — APPROVED** [review]
bingus

**Pull request merged**